doc/whats_new/v0.23.rst

```diff
@@ -473,6 +473,9 @@ Changelog
   matrix from a pandas DataFrame that contains only `SparseArray`s.
   :pr:`16728` by `Thomas Fan`_.
 
+- |Enhancement| :func:`utils.validation.check_array` supports pandas
+  IntegerArray. :pr:`16508` by `Thomas Fan`_.
```

Comment:
Maybe this could be a bit more descriptive: `utils.validation.check_array` accepts pandas `IntegerArray` when `force_all_finite` is set to `False` or `allow-nan` in which case the data is converted to `float64` values with `pd.NA` values are replaced by `np.nan`.

Maybe you could also mention that, as a consequence, all `sklearn.preprocessing` transformers (also known as scalers) that accept numeric inputs with missing values represented as `np.nan` now also accept being directly fed with pandas dataframes with `pandas.Int*` or `pandas.UInt*`- typed columns that use `pandas.NA` as the missing value marker.